druntime/src/importc_msvc_builtins_c.h

```diff
@@ -0,0 +1,183 @@
+/* This file contains C reimplementations of some of the intrinsics recognised
+   by the MSVC compiler, for ImportC.
+   This header complements the __builtins_msvc D module.
+   This header is intended for only internal use, hence the leading double underscore.
+
+   Copyright: Copyright D Language Foundation 2024-2024
+   License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+   Authors: Harry Gillanders
+   Source: $(DRUNTIMESRC importc_msvc_builtins_c.h) */
+
+#if defined(__clang__)
+#define __assume(expression) __builtin_assume(expression)
+#elif defined(__GNUC__)
+#define __assume(expression) do {if (!(expression)) {__builtin_unreachable();}} while (0)
+#else
+#define __assume(expression) __check(expression)
+#endif
+
+#ifdef __GNUC__
+#ifdef __x86_64__
+
+#ifdef __ILP32__
+#define _import_c_msvc_ptr_int unsigned int
+#else
+#define _import_c_msvc_ptr_int unsigned long long
+#endif
+
+static inline unsigned char __readgsbyte(unsigned int Offset)
+{
+    return *(unsigned char __seg_gs *) (_import_c_msvc_ptr_int) Offset;
```

Comment:
ImportC does not recognize __seg_gs or __seg_fs, so I don't see how this can work

Reply:
I thought that GDC delegated to GCC for C files, but I was mistaken. I've replaced this with inline asm in D, as is already done for DMD.